internal/controller/numaflowcontrollerrollout/numaflowcontrollerrollout_controller.go

```diff
@@ -269,539 +226,313 @@ func (r *NumaflowControllerRolloutReconciler) reconcile(
-	}
-	reconciliationResult := gitopsSync.Reconcile(targetObjs, liveObjByKey, namespace, infoProvider)
-
-	// Ignore `status` field for all comparison.
-	// TODO: make it configurable
-	overrides := map[string]sync.ResourceOverride{
-		"*/*": {
-			IgnoreDifferences: sync.OverrideIgnoreDiff{JSONPointers: []string{"/status"}}},
-	}
+// TODO: do we need to also check the underlying Deployment?
+// determine if the NumaflowController, including its underlying Deployment, has been reconciled
+// so, this requires:
+// 1. NumaflowController.Status.ObservedGeneration == NumaflowController.Generation
+// 2. Deployment.Status.ObservedGeneration == Deployment.Generation
+// 3. Deployment.Status.UpdatedReplicas == Deployment.Spec.Replicas
+func (r *NumaflowControllerRolloutReconciler) isNumaflowControllerReconciled(ctx context.Context, numaflowController *unstructured.Unstructured) (bool, string, error) {
+	numaLogger := logger.FromContext(ctx)
 
-	resourceOps, cleanup, err := r.getResourceOperations()
+	numaflowControllerStatus, err := kubernetes.ParseStatus(numaflowController)
 	if err != nil {
-		return gitopsSync.ReconciliationResult{}, nil, err
-	}
-	defer cleanup()
-
-	diffOpts := []diff.Option{
-		diff.WithLogr(*numaLogger.LogrLogger),
-		diff.WithServerSideDiff(true),
-		diff.WithServerSideDryRunner(diff.NewK8sServerSideDryRunner(resourceOps)),
-		diff.WithManager(common.SSAManager),
-		diff.WithGVKParser(clusterCache.GetGVKParser()),
+		return false, "", fmt.Errorf("failed to parse Status from NumaflowController CR: %+v, %v", numaflowController, err)
 	}
+	numaLogger.Debugf("numaflowController status: %+v", numaflowControllerStatus)
 
-	diffResults, err := sync.StateDiffs(reconciliationResult.Target, reconciliationResult.Live, overrides, diffOpts)
+	var nfcStatus apiv1.NumaflowControllerStatus
+	err = util.StructToStruct(numaflowControllerStatus, &nfcStatus)
 	if err != nil {
-		return reconciliationResult, nil, err
+		return false, "", fmt.Errorf("failed to convert NumaflowController Status: %+v, %v", numaflowController, err)
 	}
 
-	return reconciliationResult, diffResults, nil
-}
+	healthyChildCond := nfcStatus.GetCondition(apiv1.ConditionChildResourceHealthy)
 
-// getResourceOperations will return the kubectl implementation of the ResourceOperations
-// interface that provides functionality to manage kubernetes resources. Returns a
-// cleanup function that must be called to remove the generated kube config for this
-// server.
-func (r *NumaflowControllerRolloutReconciler) getResourceOperations() (kubeUtil.ResourceOperations, func(), error) {
-	clusterCache, err := r.stateCache.GetClusterCache()
-	if err != nil {
-		return nil, nil, fmt.Errorf("error getting cluster cache: %w", err)
-	}
+	numaflowControllerReconciled := numaflowController.GetGeneration() <= numaflowControllerStatus.ObservedGeneration &&
+		healthyChildCond.Status == metav1.ConditionTrue && healthyChildCond.Reason != apiv1.ProgressingReasonString
```

Comment:
```suggestion
        ncProgressing := healthyChildCond.Reason == apiv1.ProgressingReasonString
	numaflowControllerReconciled := numaflowController.GetGeneration() <= numaflowControllerStatus.ObservedGeneration && !ncProgressing
```

Reply:
Otherwise, we are setting `numaflowControllerReconciled=true` for degraded state as well

Reply:
Once complete, do you mind testing to make sure the pipeline is remaining paused until the new Controller pod is created? You can run e2e locally and have one terminal running a watch on “kubectl get pipeline” and another terminal running a watch on “kubectl get pod”